pkg/healthcheck/healthcheck.go

```diff
@@ -10,11 +10,70 @@ import (
 	"time"
 
 	acsapi "github.com/Azure/acs-engine/pkg/api"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/tools/clientcmd/api/latest"
+	"k8s.io/client-go/tools/clientcmd/api/v1"
 
+	"github.com/jim-minter/azure-helm/pkg/checks"
 	"github.com/jim-minter/azure-helm/pkg/config"
 )
 
+// GetKubeconfigFromV1Config takes a v1 config and returns a kubeconfig
+func getKubeconfigFromV1Config(kc *v1.Config) (clientcmd.ClientConfig, error) {
+	var c api.Config
+	err := latest.Scheme.Convert(kc, &c, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeconfig := clientcmd.NewDefaultClientConfig(c, &clientcmd.ConfigOverrides{})
+
+	return kubeconfig, nil
+}
+
+// HealthCheck function to verify cluster health
 func HealthCheck(ctx context.Context, cs *acsapi.ContainerService, c *config.Config) error {
+	kubeconfig, err := getKubeconfigFromV1Config(c.AdminKubeconfig)
+	if err != nil {
+		return err
+	}
+
+	restconfig, err := kubeconfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	t, err := rest.TransportFor(restconfig)
+	if err != nil {
+		return err
+	}
+
+	ctxHealthz, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
use the passed in context.  The point of context is the caller gets to decide how long the total timeout is.

Reply:
Ok.